src/app_service/service/devices/service.nim

```diff
@@ -1,66 +1,100 @@
 import NimQml, json, sequtils, system, chronicles
 
-import ./dto/device as device_dto
+import std/os
+
+import ./dto/installation as Installation_dto
+import ./dto/local_pairing_event
+import ./dto/local_pairing_status
+
 import ../settings/service as settings_service
+import ../accounts/service as accounts_service
 
+import ../../../app/global/global_singleton
 import ../../../app/core/signals/types
 import ../../../app/core/eventemitter
+import ../../../app/core/tasks/[qt, threadpool]
 import ../../../backend/installations as status_installations
+import ../../common/string_utils as string_utils
+import ../../../constants as main_constants
+
+import status_go
+
+export Installation_dto
+export local_pairing_event
+export local_pairing_status
 
-export device_dto
+include async_tasks
 
 logScope:
   topics = "devices-service"
 
 type
-  UpdateDeviceArgs* = ref object of Args
-    deviceId*: string
-    name*: string
-    enabled*: bool
+  UpdateInstallationArgs* = ref object of Args
+    installation*: InstallationDto
 
 # Signals which may be emitted by this service:
 const SIGNAL_UPDATE_DEVICE* = "updateDevice"
+const SIGNAL_LOCAL_PAIRING_EVENT* = "localPairingEvent"
+const SIGNAL_LOCAL_PAIRING_STATUS_UPDATE* = "localPairingStatusUpdate"
 
 QtObject:
   type Service* = ref object of QObject
     events: EventEmitter
+    threadpool: ThreadPool
     settingsService: settings_service.Service
+    accountsService: accounts_service.Service
+    localPairingStatus: LocalPairingStatus
```

Comment:
@igor-sirotin wouldn't be better if we could do a lazy initialization here? (having this `nil` at the beginning and initialize it only if there is a need for that)